helix-term/src/commands.rs

```diff
@@ -5100,3 +5103,39 @@ fn rename_symbol(cx: &mut Context) {
     );
     cx.push_layer(Box::new(prompt));
 }
+
+/// Increment object under cursor by count.
+fn increment(cx: &mut Context) {
+    increment_impl(cx, cx.count() as i64);
+}
+
+/// Decrement object under cursor by count.
+fn decrement(cx: &mut Context) {
+    increment_impl(cx, -(cx.count() as i64));
+}
+
+/// Decrement object under cursor by `amount`.
+fn increment_impl(cx: &mut Context, amount: i64) {
+    let (view, doc) = current!(cx.editor);
+    let selection = doc.selection(view.id);
+    let text = doc.text();
+
+    let changes = selection.ranges().iter().filter_map(|range| {
+        NumberIncrementor::from_range(text.slice(..), *range).map(|incrementor| {
+            let new_text = incrementor.incremented_text(amount);
+            (
+                incrementor.range.from(),
+                incrementor.range.to(),
+                Some(new_text),
+            )
+        })
```

Comment:
This pattern sure is weird but it does work.

Reply:
Yeah, there are times when using `map` on an `Option` makes code nicer, but in this case it just makes it harder to read and increases the level of indentation. I'll change it.